lib/ansible/module_utils/artifactory.py

```diff
@@ -205,6 +205,23 @@ def get_always_required_keys(self, url, config_dict):
                                 " list." % config_value)
         return req_keys
 
+    def compare_config(self, current, desired):
+        def order_dict_sort_list(dictionary):
+            result = {}
+            for k, v in sorted(dictionary.items()):
+                if isinstance(v, dict):
+                    result[k] = order_dict_sort_list(v)
+                elif isinstance(v, list):
+                    result[k] = sorted(v)
+                else:
+                    result[k] = v
+            return result
+
+        s_current = order_dict_sort_list(current)
+        s_desired = order_dict_sort_list(desired)
+        return all(s_current[k] == s_desired[k]
+                   for k in s_desired if k in s_current)
```

Comment:
So, this is the hitch. It will only compare keys IF they EXIST in configuration pulled from the remote. Not all keys will be returned every time (such as password never being returned when creating a user).

So if a user has a typo, it won't apply that field. This is sort've the consequence of doing a deep comparison. Without explicitly saying "Hey, that value is not allowed", which would require manually creating the correct config for each module (I see where this is going...), we can only compare what is in both.